### ISSUE TITLE
Create a storage_account_network_rules resource

### ISSUE THREAD

**Issue by MattMencel** [terraform-providers/terraform-provider-azurerm#3522]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Create a `azurerm_storage_account_network_rules` resource so they can be managed separately from the `azurerm_storage_account` resource.

### New or Affected Resource(s)

* azurerm_storage_account_network_rules
* azurerm_storage_account

### Potential Terraform Configuration

```hcl
resource "azurerm_storage_account_network_rules" "network_rules" {
  name = "network_rules
  storage_account = "${azurerm_storage_account.storage_account.id}"
  bypass = "None"
  ip_rules = "${var.myip}"
  virtual_network_subnet_ids = ["${var.my_vnet_list}"]
}
```

**Comment by lawrenae** [comment]
we were looking into this (or similar) recently at work -- is what this is proposing effectively different than https://www.terraform.io/docs/providers/azurerm/r/storage_account.html#network_rules ?

**Comment by MattMencel** [comment]
I'm proposing a separate resource if possible.  Similar to the way it works with `azurerm_sql_firewall_rule`.  I've got a customer that would like to manage the network_rules outside of the storage_account resource.

**Comment by evenh** [comment]
This functionality is very much needed, thanks for reporting and opening up a PR @MattMencel.

**Comment by derekrprice** [comment]
See #4186 for why this is needed.  Currently, a storage account can't be created that allows access from the IPs from a Function App without creating a circular reference that Terraform can't resolve.

**Comment by geertn** [comment]
Not sure if this affects your work but please take note of:
https://github.com/terraform-providers/terraform-provider-azurerm/issues/4574

Just a FYI

**Comment by tombuildsstuff** [comment]
Fixed via #5082
